distributed/executor.py

```diff
@@ -1798,3 +1829,62 @@ def futures_of(o, executor=None):
             raise CancelledError(bad)
 
     return list(futures)
+
+
+@contextmanager
+def serialize_for_publish(x):
```

Comment:
The idea was to make a serializable version temporarily within the context manager. There are probably some nice functions in your util code to walk around dask and dask-like objects; but things already persisted should tend to be shallow and easy to work with.